src/jllvm/vm/JIT.hpp

```diff
@@ -82,6 +79,41 @@ class JIT
             createLambdaMaterializationUnit(std::move(symbol), m_optimizeLayer, f, m_dataLayout, m_interner));
     }
 
+    /// Add all symbol-implementation pairs to the implementation library.
+    /// The implementation library contains implementation of functions used by the materialization
+    /// (bytecode compiler, JNI bridge, etc.).
+    template <class Ss, class... Fs>
+    void addImplementationSymbols(std::pair<Ss, Fs>&&... args)
+    {
+        (addImplementationSymbol(std::move(args.first), std::move(args.second)), ...);
+    }
+
+    /// Add callable 'f' as implementation for symbol 'symbol' to the implementation library.
+    template <class F>
+    void addImplementationSymbol(std::string symbol, const F& f)
```

Comment:
Does `symbol` have to be a `std::string` because `createLambdaMaterializationUnit` cannot be changed to accommodate non owning references?

Reply:
After double checking, yes pretty much, and using `std::string` as parameter is the easiest effiecent version (just `std::move`ing it along).